ivy.el

```diff
@@ -4578,10 +4578,12 @@ Otherwise, forward to `ivy-kill-line'."
     x))
 
 (defun ivy--action-insert (x)
-  (insert
-   (if (stringp x)
-       (ivy--trim-grep-line-number x)
-       x (car x))))
+  (insert (cond ((stringp x) (ivy--trim-grep-line-number x))
+                ;; swiper-isearch passes buffer pos
+                ((numberp x) (save-excursion
+                               (goto-char x)
+                               (buffer-substring (line-beginning-position) (line-end-position))))
+                (t x (car x)))))
```

Comment:
AKA `(cond ... (t (car x)))` or even `(cond ... ((car x)))`.

Reply:
The original `if` had a standalone `x` in the `then` clause that I was unusure about removing, but with your blessing I will. :)

Reply:
Although I guess it's moot if I'm fixing this in swiper.

Reply:
> The original `if` had a standalone `x` in the `then` clause

Looks like a typo to me, as it's a no-op and the indentation is off.